File: apps/auth-server/src/main/java/com/example/chat/auth/server/api/dto/request/AuthenticateRequest.java
package com.example.chat.auth.server.api.dto.request;

import jakarta.validation.constraints.NotBlank;

/**
 * 인증 요청 DTO
 */
public record AuthenticateRequest(
        @NotBlank(message = "identifier cannot be blank") String identifier,
        @NotBlank(message = "credentialType cannot be blank") String credentialType,
        @NotBlank(message = "credentialData cannot be blank") String credentialData) {
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/api/dto/request/AuthenticateRequest.kt
package com.example.chat.auth.server.api.dto.request

import jakarta.validation.constraints.NotBlank

/** 인증 요청 DTO */
data class AuthenticateRequest(
        @field:NotBlank(message = "identifier cannot be blank") val identifier: String? = null,
        @field:NotBlank(message = "credentialType cannot be blank")
        val credentialType: String? = null,
        @field:NotBlank(message = "credentialData cannot be blank")
        val credentialData: String? = null
)
